qiita_plugins/target_gene/tgp/qiita_client.py

```diff
@@ -0,0 +1,116 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from os.path import join, dirname, abspath
+from os import environ
+from future import standard_library
+
+import requests
+
+with standard_library.hooks():
+    from configparser import ConfigParser
+
+
+class QiitaClient(object):
+    """Client of the Qiita RESTapi
+
+    Parameters
+    ----------
+    server_url : str
+        The url of the Qiita server
+
+    Methods
+    -------
+    get
+    post
+    """
+    def __init__(self, server_url):
+        self._server_url = server_url
+
+        try:
+            conf_fp = environ['QP_TARGET_GENE_CONFIG_FP']
+        except KeyError:
+            conf_fp = join(dirname(abspath(__file__)), 'support_files',
+                           'config_file.cfg')
+
+        config = ConfigParser()
+        with open(conf_fp, 'U') as conf_file:
+            config.readfp(conf_file)
+
+        server_cert = config.get('main', 'SERVER_CERT')
+        if not server_cert:
+            # The server certificate is not provided, use standard certificate
+            # verification methods
+            self._verify = True
```

Comment:
Yeah, I hate this - however this is a limitation from `requests` (see documentation here: `http://docs.python-requests.org/en/latest/user/advanced/#ssl-cert-verification`). By setting up the `verify` parameter here, I'm avoiding the execution of the if statement every time a request is done.